web/src/main/java/org/wso2/testgrid/web/sso/TestgridSSOAgentFilter.java

```diff
@@ -0,0 +1,119 @@
+ * under the License.
+ */
+
+package org.wso2.testgrid.web.sso;
+
+import org.apache.commons.lang.StringUtils;
+import org.glassfish.jersey.internal.util.Base64;
+import org.wso2.carbon.identity.sso.agent.SSOAgentConstants;
+import org.wso2.carbon.identity.sso.agent.SSOAgentFilter;
+import org.wso2.carbon.identity.sso.agent.bean.SSOAgentConfig;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+
+/**
+ * Implementation of the SSOAgent Filter which identifies the correct httpBinding
+ * to send the request to Identity Provider.
+ */
+public class TestgridSSOAgentFilter extends SSOAgentFilter {
+
+    private static Logger logger = Logger.getLogger("org.wso2.testgrid.web.sso.TestgridSSOAgentFilter");
+
+    private static final String USERNAME = "username";
+    private static final String PASSWORD = "password";
+    private static final String CHARACTER_ENCODING = "UTF-8";
+    private static Properties properties;
+    private FilterConfig filterConfig = null;
+
+    static {
+        properties = SSOContextEventListener.getProperties();
+    }
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        this.filterConfig = filterConfig;
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
+                         FilterChain filterChain) throws IOException, ServletException {
+
+        String httpBinding = servletRequest.getParameter(
+                SSOAgentConstants.SSOAgentConfig.SAML2.HTTP_BINDING);
+        if (httpBinding != null && !httpBinding.isEmpty()) {
+            if ("HTTP-POST".equals(httpBinding)) {
+                httpBinding = "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST";
+            } else if ("HTTP-Redirect".equals(httpBinding)) {
+                httpBinding = "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect";
+            } else {
+                logger.log(Level.INFO, "Unknown SAML2 HTTP Binding. Defaulting to HTTP-POST");
+                httpBinding = "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST";
+            }
+        } else {
+            logger.log(Level.INFO, "SAML2 HTTP Binding not found in request. Defaulting to HTTP-POST");
+            httpBinding = "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST";
+        }
+        SSOAgentConfig config = (SSOAgentConfig) filterConfig.getServletContext().
+                getAttribute(SSOAgentConstants.CONFIG_BEAN_NAME);
+        config.getSAML2().setHttpBinding(httpBinding);
+
+        if (StringUtils.isNotEmpty(servletRequest.getParameter(USERNAME)) &&
+                StringUtils.isNotEmpty(servletRequest.getParameter(PASSWORD))) {
```

Comment:
Why do you do this check? Can you move this logic to a separate method? Then, the method name will document what you r actually trying to do.

Reply:
Logic is taken to separate method.
 
This is kept for the future purpose of providing a Login Page in our app. For a user who will use that form will be having a request with these parameters. The way we have to handle such request is implemented inside this block.

Reply:
I'm sorry, I meant to ask you to move the logic in `if` condition to a separate method. But, +1 for what u moved as well.